src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -49,7 +50,11 @@ def download_coverage_artifacts(build_task_id):
 
 def get_github_commit(mercurial_commit):
     url = 'https://api.pub.build.mozilla.org/mapper/gecko-dev/rev/hg/%s'
-    r = requests.get(url % mercurial_commit)
+    while True:
```

Comment:
Yes, I agree with you regarding the problems with the mapper. It isn't a large problem for now though, as we are only running the task once per day with a build from the previous day. So the only time this can fail is when there's some kind of problem (in the past few weeks, it's only happened once).

> Another option could also be to generate an html coverage report and push it to some s3 bucket and have data for all commits there.

> Then have some frontend (elm) code that would show you the commits and their coverage. This way we wouldn't have to use other services. but yes, this requires more work.

Yeah, this requires quite a bit of work. So we initially decided to rely on an external service to deal with it.